### PR TITLE
Site Level User Profile: use site-level language (and admin color) on Calypso pages

### PR DESCRIPTION
**Pull request by fushar**
> [!CAUTION]
> DO NOT MERGE this PR yet until the Jetpack [PR](https://github.com/Automattic/jetpack/pull/38879) is deployed to Atomic. (But you can still review.)

Related to:

- https://github.com/Automattic/dotcom-forge/issues/7754

## Proposed Changes

This PR makes site-specific Calypso pages to use the site-level user language which set in `wp-admin/profile.php`.

The logic is very similar to:

- https://github.com/Automattic/wp-calypso/pull/89261

This PR makes use of the newly created `/wpcom/v2/profile` endpoint, which returns the user's locale (along with the admin color). Then, it sets the current locale based on the response.

Later, we will clean up the now-unused `/wpcom/v2/admin-color` endpoint as it's merged to the above new endpoint.

|Type|Before|After|
|-|-|-|
|Atomic<br>Classic|image|image|
|Simple<br>Classic|image|image|
|Atomic<br>Default|image|image|
|Simple<br>Default|image|image

(Atomic Default is pending https://github.com/Automattic/dotcom-forge/issues/8511).
## Why are these changes being made?

We want to expose `profile.php` settings to all types of sites.

## Testing Instructions

> [!WARNING]
> !!! IMPORTANT: USE THE CALYPSO LIVE URL FROM THIS PR INSTEAD !!!:
> - https://github.com/Automattic/wp-calypso/pull/93733

### Test on a Simple Classic site

1. Prepare a Simple Classic site.
8. Go to your site's `/wp-admin/profile.php`:
    - Change the admin color to a color other than the one you set in `/me/account`.
    - Change the user language to a language other than the one you set in `/me/account`.
10. Go to `<calypso live URL>/home/<site slug>`:
    - VERIFY that the color is the same as the color you set previously.
    - VERIFY that the language is the same as the language you set in step 8.
1. Don't do anything on your sandbox, and perform the next sections below.

### Test on a Simple Default site

1. Prepare a Simple Default site.
1. Go to `<calypso live URL>/home/<site slug>`:
    - VERIFY that it's still using global admin color and langauge.
1. Now, on your sandbox, run:
    - `wpsh`
    - `update_blog_option( <your blog id>, 'wpcom_site_level_user_profile', '1' );`
1. Follow the same instructions as above from step (4) onwards.

### Test on an Atomic Classic site

1. Prepare an Atomic Classic site.
1. Use Bleeding Edge for the Jetpack plugin via Jetpack Beta Tester, to pick up this PR https://github.com/Automattic/jetpack/pull/38879
1. Follow the same instructions as above from step (4) onwards.

### Test on an Atomic Default site

1. Prepare an Atomic Default site.
1. Use Bleeding Edge for the Jetpack plugin via Jetpack Beta Tester, to pick up this PR https://github.com/Automattic/jetpack/pull/38879
2. Go to `_cli` then run `wp option update wpcom_site_level_user_profile 1`
1. Follow the same instructions as above from step (4) onwards.

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by okmttdhr** [comment]
I tested it with Simple Classic, and I noticed something weird;

- Change the admin color to a color other than the one you set in /me/account.
- Change the user language to a language other than the one you set in /me/account.
- Go to <calypso live URL>/home/<site slug>:
- ✅ VERIFY that the color is the same as the color I set.
- ✅ VERIFY that the language is the same as the language I set.
- Then, I clicked the W icon to go to /sites
- ❌ The language and the color schema were the same as I just set for the site (not a global one). If I reload the page, they become the global color scheme & language.

**Comment by okmttdhr** [comment]
These might be out of the scope of this PR;

There is a slight discrepancy between the translations in Calypso and wp-admin, likely due to differences in translation context. (The Japanese translation of "新着" feels a bit off, BTW: it usually conveys the meaning like "news" or "new arrivals")

| Calypso | Header |
|--------|--------|
| Screenshot 2024-08-16 at 11 30 36 | Screenshot 2024-08-16 at 11 30 42 |
| Screenshot 2024-08-16 at 11 31 02 | Screenshot 2024-08-16 at 11 31 07 | 

Additionally, we can see the different colors in the submenu (it should be a different issue).

**Comment by fushar** [comment]
> There is a slight discrepancy between the translations in Calypso and wp-admin, likely due to differences in translation context. 

I'm seeing in Indonesian too 😅 not sure what to do, though.

> Additionally, we can see the different colors in the submenu (it should be a different issue).

Yes... I think this is captured in this issue: https://github.com/Automattic/dotcom-forge/issues/8470

**Comment by fushar** [comment]
> Then, I clicked the W icon to go to /sites
> ❌ The language was the same as I just set for the site (not a global one). If I reload the page, they become the global language.

I can reproduce this on Calypso Live URL but not locally 🫠 I'll continue debugging, thanks!

**Comment by fushar** [comment]
@okmttdhr I fixed the bleeding local -> global locale issue in a different PR so that I can get more thorough review:

- https://github.com/Automattic/wp-calypso/pull/93733

Could you retest? I think I can't repro your issues anymore.

**Comment by fushar** [comment]
I added the Atomic Default case. It still has mixed languages in the sidebar (see screenshot). This is really weird... I ran out of time debugging this 😭

**Comment by fushar** [comment]
> It still has mixed languages in the sidebar (see screenshot).

Ah, I think it's caused by this line. We should gate this by the `wpcom_site_level_user_profile` flag:

https://github.com/Automattic/jetpack/blob/a950c3f0172479cd01e8a55ad030bb309684b80f/projects/packages/masterbar/src/masterbar/class-masterbar.php#L143

@okmttdhr, can I leave this to you 🙏 after that all cases should now work fine!!

File: client/layout/index.jsx [hunk 1]
```diff
@@ -265,6 +272,10 @@ class Layout extends Component {
 		) {
 			this.refreshColorScheme( prevProps.colorScheme, this.props.colorScheme );
 		}
+
+		if ( prevProps.locale !== this.props.locale ) {
```
Comment (miksansegundo) [inline client/layout/index.jsx:276]: I haven't tested this code but it might be worth checking that `this.props.locale` is not `null` or and empty string because in other cases we found that WordPress uses `null` or and empty string for English.

**Comment by okmttdhr** [comment]
📝 This PR shouldn't be needed due to the discussion: p1724217941726209/1724171890.507159-slack-CRWCHQGUB
TLDR: We're to re-tangle the locale setting to /me/account.

**Comment by fushar** [comment]
> This PR shouldn't be needed

Yes. At least, I will create another PR to just use the new `/wpcom/v2/profile` endpoint. It's much more concise than the current `/wpcom/v2/admin-color endpoint, which is duplicated across Simple and Jetpack codebase.
